crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -379,30 +379,30 @@ async fn price_improvement_fee_sell_order_test(web3: Web3) {
     // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
     //
     // Quote: 10000000000000000000 GNO for 9871580343970612988 DAI with
-    // 294580438010728 GNO fee. Equivalent to: (10000000000000000000 +
-    // 294580438010728) GNO for 9871580343970612988 DAI, then scaled to sell amount
-    // gives 10000000000000000000 GNO for 9871289555090525964 DAI
+    // 294580438010728 GNO fee. Equivalent to: 10000000000000000000 GNO for
+    // 9871580343970612988 * ((10000000000000000000 - 294580438010728) /
+    // 10000000000000000000) = 9871289546524454223 DAI
     //
-    // Price improvement over quote: 9871415430342266811 - 9871289555090525964 =
-    // 125875251741847 DAI. Protocol fee = 0.3 * 125875251741847 DAI =
-    // 37762575522554 DAI
+    // Price improvement over quote: 9871415430342266811 - 9871289546524454223 =
+    // 125883817812588 DAI. Protocol fee = 0.3 * 125883817812588 DAI =
+    // 37765145343776 DAI
     //
-    // Protocol fee in sell token: 37762575522554 DAI / 9871415430342266811 *
-    // (10000000000000000000 - 167058994203399) = 38253829890184 GNO
+    // Protocol fee in sell token: 37765145343776 DAI / 9871415430342266811 *
+    // (10000000000000000000 - 167058994203399) = 38256433142280 GNO
     //
     // Final execution is 10000000000000000000 GNO for (9871415430342266811 -
-    // 37762575522554) = 9871377667766744257 DAI, with 205312824093583 GNO fee
+    // 37765145343776) = 9871377665196923035 DAI, with 205315427345679 GNO fee
     //
-    // Settlement contract balance after execution = 205312824093583 GNO =
-    // 205312824093583 GNO * 9871377667766744257 / (10000000000000000000 -
-    // 205312824093583) = 202676203868731 DAI
+    // Settlement contract balance after execution = 205315427345679 GNO =
+    // 205315427345679 GNO * 9871377665196923035 / (10000000000000000000 -
+    // 205315427345679) = 202678773689953 DAI
     execute_test(
         web3.clone(),
         vec![ProtocolFeesConfig(vec![protocol_fee])],
         OrderKind::Sell,
         None,
-        205312824093583u128.into(),
-        202676203868731u128.into(),
+        205315427345679u128.into(),
+        202678773689953u128.into(),
```

Comment:
Yes, very clear that this is now working as intended ;-)

Reply:
That will be tackled in the https://github.com/cowprotocol/services/pull/2561 🙂